src/compiler/checker.ts

```diff
@@ -31378,14 +31379,15 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     // Lookup the private identifier lexically.
-    function lookupSymbolForPrivateIdentifierDeclaration(propName: __String, location: Node): Symbol | undefined {
-        for (let containingClass = getContainingClass(location); !!containingClass; containingClass = getContainingClass(containingClass)) {
+    function lookupSymbolForPrivateIdentifierDeclaration(propName: __String, location: Node, containingClass = getContainingClassExcludingClassDecorators(location)): Symbol | undefined {

```

Comment:
I'm not sure I like passing in `containingClass` here as it results in `location` being unused.

Instead, we could make the third parameter something like `includeLocation?: boolean`, pass in the containing class you already resolved as `location`, and tweak the original `for` loop to use `location` if `includeLocation` is true and it's already a `ClassLike`.

Reply:
Either that or drop `location` and require callers always pass in `containingClass` themselves.